rosidl_default_generators/rosidl_default_generators-extras.cmake.in

```diff
@@ -1,15 +1,14 @@
 # generated from
 # rosidl_default_generators/rosidl_default_generators-extras.cmake.in
 
+find_package("ament_cmake_core" REQUIRED)
+ament_index_get_resources(rosidl_typesupport_c_packages "rosidl_typesupport_c")
+ament_index_get_resources(rosidl_typesupport_cpp_packages "rosidl_typesupport_cpp")
 set(_exported_dependencies
   "rosidl_typesupport_c"
   "rosidl_typesupport_cpp"
-  "rosidl_typesupport_introspection_c"
-  "rosidl_typesupport_introspection_cpp"
-  "rosidl_typesupport_connext_c"
-  "rosidl_typesupport_connext_cpp"
-  "rosidl_typesupport_opensplice_c"
-  "rosidl_typesupport_opensplice_cpp"
```

Comment:
If we keep the dependencies in the package.xml, I think we should keep them here too, shouldn't we ?

Reply:
I don't see a reason why. Using the index makes this logic fully "dynamic" if you e.g. don't have one of them the following code would anyway skip them (since `find_package` won't be able to find the package).

Reply:
Yeah that's fine as they already register themselves in the index as type support packages